### PR TITLE
[14.0][ADD] purchase_operating_unit_access_all

### PR DESCRIPTION
**Pull request by kittiu**
This module allow a user to have Access all OUs' purchase orders, without having to add OUs in user setting.

Example use case, a shared procurement team under an OU but need to work on purchase orders of all OUs.

To configure a user to have access to all OUs' purchase orders (without all OUs access):

* Go to *Settings / Users & Companies / Users*
* For a user, select checkbox "Access all OUs' purchase orders"

### CODE REVIEW

**Comment by kittiu** [comment]
Ref: https://github.com/OCA/operating-unit/pull/383

File: purchase_operating_unit_access_all/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
```
Comment (newtratip) [inline purchase_operating_unit_access_all/__init__.py]:
```suggestion
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
```

File: purchase_operating_unit_access_all/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2020 Ecosoft Co., Ltd. (http://ecosoft.co.th)
```
Comment (newtratip) [inline purchase_operating_unit_access_all/__manifest__.py]:
```suggestion
# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
```

File: purchase_operating_unit_access_all/readme/CONFIGURE.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (newtratip) [inline purchase_operating_unit_access_all/readme/CONFIGURE.rst:1]:
```suggestion
```

File: purchase_operating_unit_access_all/security/purchase_security.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <field name="name">Access all OUs' purchase orders</field>
+        <field name="category_id" ref="base.module_category_usability" />
+    </record>
+
+    <function model="ir.rule" name="write">
+        <value
+            eval="[ref('purchase_operating_unit.ir_rule_purchase_order_allowed_operating_units'), ref('purchase_operating_unit.ir_rule_purchase_order_line_allowed_operating_units')]"
+        />
+        <value
+            eval="{'domain_force': &quot;['|','|',(1, '=', user.has_group('purchase_operating_unit.group_all_ou_purchase')),('operating_unit_id','=',False),('operating_unit_id','in', user.operating_unit_ids.ids)]&quot;}"
```
Comment (newtratip) [inline purchase_operating_unit_access_all/security/purchase_security.xml]:
```suggestion
            eval="{'domain_force': &quot;['|','|',(1, '=', user.has_group('purchase_operating_unit_access_all.group_all_ou_purchase')),('operating_unit_id','=',False),('operating_unit_id','in', user.operating_unit_ids.ids)]&quot;}"
```

Comment (kittiu) [reply purchase_operating_unit_access_all/security/purchase_security.xml]: Thanks!

**Comment by newtratip** [comment]
I'm not in Access all OUs' purchase orders group. I'm going to purchase order menu, it's throw error as below
Selection_098

**Comment by kittiu** [comment]
> I'm not in Access all OUs' purchase orders group. I'm going to purchase order menu, it's throw error as below

Thanks!

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM!

**Pull request merged**